### ISSUE TITLE
imwrite with partially-filled single tiles in contiguous mode creates corrupted files

### ISSUE THREAD

**Issue by jmuhlich** [cgohlke/tifffile#3]
If `tile` is set, the image shape is smaller than `tile`, and `contiguous` is True, then the number of bytes allocated and written is less than the value in TileByteCounts. Furthermore there is no way to disable contiguous mode as the `contiguous` argument to `imwrite` is unconditionally overwritten with True. Here is a minimal reproduction:
```python
import numpy as np
import tifffile
img = np.zeros((15, 15), np.uint8)
# This vertical stripe makes the problem stand out in the image data.
img[:, 0] = 255
tifffile.imsave('tiled-15x15.tif', img, tile=(16, 16))
```
The output file in my case contains `TileOffsets <320>` and `TileByteCounts <256>` but the file is only 545 bytes long when it should be 320+256=576. From looking at a hex dump of the file it can be seen that only 15\*15 bytes from `img` were written out without padding to 16\*16. tifffile will actually read this back in correctly, but every other tool I tried either reports it as corrupt or reads the image data as if it were padded and ignores the buffer overrun (yielding an image with a diagonal stripe instead of a vertical one and possible garbage at the bottom).

I traced this to the "contiguous" support introduced all the way back in v0.13.1 (2017-something?) and tried to disable that by passing `contiguous=False`, but I think there is also a small logic bug that prevents this from working. tifffile.py:1658 (in current master) says `contiguous = not compress` where I think it should say `contiguous = contiguous and not compress`.

I'll be submitting a PR to address the `contiguous` problem, but I'm not sure what the use case for contiguous mode is so I'm hesitant to propose a fix for the bigger problem of corrupted output.

**Comment by cgohlke** [comment]
Thank you. Indeed, contiguous mode should be disabled automatically for single tiles that are larger than the image data. This will be fixed in the next version.

**Comment by cgohlke** [comment]
Fixed in v2020.5.7
